doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+> `tab<Property>` style here, for comparison. At the time of writing, I'm unsure
+> of which is better. I'll be using `<element>.<property>` for the remainder of
+> the doc.
+
+In the above settings snippet, we see the following things:
+1. A list of `themes` that the user can pick from. Each theme has a `name`
+   property used to identify the theme, and a group of properties for the theme.
+2. The user has set the `theme` to `"My Boxy Theme"`, the first theme
+   in the list of themes. If the user wanted to switch to the other installed
+   theme, `"My small light theme"`, they'd simply need to change this property.
+
+### Exposed theme properties
+
+Themes should be able to control a variety of elements of the Terminal UI. Some
+of these settings will be easier to implement than others. As such, below is a
+set of properties that seems appropriate to include as part of a "v1" theming
+implementation. In [Future Considerations](#future-considerations), we'll
+enumerate additional properties that could be added in the future to further
+control the UI.
+
+#### Theming v1 Properties
+
+* `tab.cornerRadius`: Control the radius of the corners of the tab items.
+  Accepts a `double`. If this is set to `0`, then the tabs will have squared-off
+  corners. No particular limit is set on the max value accepted, though larger
+  values might not be aesthetically pleasing.
+* `tab.closeButton`: Control the visibility of the close button for a tab item.
+  Accepts the following values:
+    - `visible`: The default behavior of the tab item close button - always
+      visible.
+    - `hover`: The close button on a tab item only appears when the tab is
+      hovered.
+    - `hidden`: The close button on a tab is always hidden.
+* `tab.icon`: Control the visibility, appearance of the tab icon
+    - `visible`: The default behavior of the tab item icon - always visible, and
+      in full color.
+    - `outline`: The icon is always visible, but is only drawn as an outline,
+      using `BitmapIconSource.ShowAsMonochrome(true)`
+    - `hidden`: The icon is hidden
+* `tab.background`: Control the color of the background of tab items. See below
+  for accepted colors.
+* `tabRow.background`: Control the color of the background of the tab row items.
+  See below for accepted colors.
+* `pane.borderColor`: Control the color of the border used to separate panes.
+  This is the color of the inactive border between panes.
+* `pane.activeBorderColor`: Control the color of the border of the active pane
+* `pane.borderWidth`: Control the width of the borders used to separate panes.
+* `window.applicationTheme`: If set, will set the XAML `RequestedTheme`
+  property. This can be one of `light`, `dark` or `system`. This controls how
+  XAML fundamentally styles UI elements. If not provided, will use the default
+  value "system", which will use whatever the system's default theme is.
+
+#### Theme Colors

```

Comment:
Hey something else we should consider - someone might want an acrylic titlebar, that's _not_ the terminal BG color. This spec doesn't cover that case quite yet.

maybe `"property": { "color": "#rrggbb", "opacity": .5 }`?